--- a/com/mojang/blaze3d/platform/GLX.java
+++ b/com/mojang/blaze3d/platform/GLX.java
@@ -69,7 +69,7 @@
 
    public static LongSupplier _initGlfw() {
       RenderSystem.assertThread(RenderSystem::isInInitPhase);
-      MainWindow.func_211162_a((p_229879_0_, p_229879_1_) -> {
+      if (!(Boolean.parseBoolean(System.getProperty("forge.7285.extra", "false")) && !Boolean.parseBoolean(System.getProperty("fml.earlyprogresswindow", "true")))) MainWindow.func_211162_a((p_229879_0_, p_229879_1_) -> {
          throw new IllegalStateException(String.format("GLFW error before init: [0x%X]%s", p_229879_0_, p_229879_1_));
       });
       List<String> list = Lists.newArrayList();
